sympy/solvers/ode/ode.py

```diff
@@ -1639,6 +1639,34 @@ def odesimp(ode, eq, func, hint):
     if not isinstance(eq, Equality):
         raise TypeError("eq should be an instance of Equality")
 
+    if eq.lhs.is_Piecewise or eq.rhs.is_Piecewise:
```

Comment:
Do ODE solutions always have an implicit assumption that a constant is not 0? If so, symbols could be created with that assumption and then the `Ne(C1, 0)` type constraints would evaluate on their own and not appear in solutions. If this is not a good assumption but is a good assumption here, you could do a `eq.subs(nonzero).subs(original)` pair of substitutions where `nonzero` are the constants mapped to non-zero dummies and `orig` are the dummies mapped to the original (perhaps vanilla-assumption) constants.

Reply:
You won't want to change the assumptions on variables by doing only the nonzero subs; you would want to restore the original (expected) symbols: `Symbol('a') != Symbol('a', positive=True)`.